### PR TITLE
bugfix: check if dir exist before trying to create one

### PR DESCRIPTION
**Pull request by Karol-Stelmaczonek**
Ticket: https://oat-sa.atlassian.net/browse/INF-212

## What's Changed

- Bugfix for error when creating new directory.

### CODE REVIEW

File: common/persistence/class.PhpFileDriver.php [hunk 1]
```diff
@@ -260,14 +260,14 @@ private function writeFile($id, $value, $preWriteValueProcessor = null)
...
+     * @param string $path
      * @param int $mode
      * @return bool
      */
     private function makeDirectory(string $path, int $mode)
     {
-        if (is_dir($path) || @mkdir($path, $mode, true)) {
-            return true;
+        if (!is_dir($path)) {
+            return @mkdir($path, $mode, true);
```
Comment (pribi) [inline common/persistence/class.PhpFileDriver.php:270]: Is it good, if mkdir is not successful (=returns false), then the makeDirectory method just returns false without logging anything?

Comment (Karol-Stelmaczonek) [reply common/persistence/class.PhpFileDriver.php]: Good point, changed it a bit

**Review by pribi — APPROVED** [review]
LGTM.

File: common/persistence/class.PhpFileDriver.php [hunk 2]
```diff
@@ -260,30 +260,29 @@ private function writeFile($id, $value, $preWriteValueProcessor = null)
 
     /**
      * Create directory and suppress warning message
-     * @param $path
+     * @param string $path
      * @param int $mode
      * @return bool
      */
     private function makeDirectory(string $path, int $mode)
```
Comment (pnal) [inline common/persistence/class.PhpFileDriver.php:267]: Could you specify return type according to a usage of that method?

Comment (Karol-Stelmaczonek) [reply common/persistence/class.PhpFileDriver.php]: added viod

File: common/persistence/class.PhpFileDriver.php [hunk 3]
```diff
@@ -260,30 +260,29 @@ private function writeFile($id, $value, $preWriteValueProcessor = null)
...
-                sprintf(
-                    'Directory was not created. File with the same name already exists. Path: \'%s\'',
-                    $path
-                )
-            );
-        } else {
-            \common_Logger::w(sprintf('Directory was not created. Path: \'%s\'', $path));
+        if (file_exists($path)) {
+            if (is_dir($path)) {
+                \common_Logger::w(sprintf('Directory already exists. Path: \'%s\'', $path));
```
[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply common/persistence/class.PhpFileDriver.php]: changed to info

**Review by pnal — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Comment by pnal** [comment]
but please fix phpcs issues

**Pull request merged**